src/rideables/SSHashTable.hpp

```diff
@@ -0,0 +1,322 @@
+#ifndef SSHASHTABLE_HPP
+#define SSHASHTABLE_HPP
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+#include <atomic>
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include <utility>
+
+#include "HarnessUtils.hpp"
+#include "ConcurrentPrimitives.hpp"
+#include "RMap.hpp"
+#include "RCUTracker.hpp"
+#include "CustomTypes.hpp"
+
+template <class K, class V>
+class SSHashTable : public RMap<K, V>
+{
+    struct Node;
+
+    struct MarkPtr
+    {
+        std::atomic<Node *> ptr;
+        MarkPtr(Node *n) : ptr(n){};
+        MarkPtr() : ptr(nullptr){};
+    };
+
+    struct Node
+    {
+        K key;
+        V val;
+        MarkPtr next;
+        Node(K k, V v, Node *n) : key(k), val(v), next(n){};
+    };
+
+private:
+    int size = 1000000; //number of buckets for hash table
+    int MAX_LOAD = 100000000;
+    padded<MarkPtr> *buckets = new padded<MarkPtr>[size] {};
```

Comment:
I don't think this would compile; `size` is not constant. Do dynamic allocation in the constructor.